extension/awsxrayproxy/config.go

```diff
@@ -0,0 +1,50 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsxrayproxy
+
+import (
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/config/configtls"
+)
+
+// Config defines the configuration for an AWS X-Ray proxy.
+type Config struct {
+	config.ExtensionSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+
+	// endpoint is the TCP address and port on which this receiver listens for
```

Comment:
Can we probably edit the description? I think in this case we will be using X-Ray service APIs like `GetSamplingRules` and `GetSamplingTargets` via this proxy right? So we probably don't need X-Ray SDK in this case.